lifetimes/estimation.py

```diff
@@ -36,6 +37,185 @@ def _print_params(self):
             s += "%s: %.2f, " % (p, value)
         return s.strip(', ')
 
+class BetaGeoBetaBinomFitter(BaseFitter):
+    """
+    Also known as the Beta-Geometric/Beta-Binomial Model Model [1].
+
+    Future purchases opportunities are treated as discrete points in time. In the literature,
+    the model provides a better fit than the Pareto/NBD model for a nonprofit organization
+    with regular giving patterns.
+
+    The model is estimated with a recency-frequency matrix with n transaction opportunities.
+
+    [1] Fader, Peter S., Bruce G.S. Hardie, and Jen Shang (2010),
+        "Customer-Base Analysis in a Discrete-Time Noncontractual Setting,"
+        Marketing Science, 29 (6), 1086-1108.
+
+    """
+    
+    def __init__(self, penalizer_coef=0.):
+        self.penalizer_coef = penalizer_coef
+
+    @staticmethod
+    def _loglikelihood(params, x, tx, T):
+
+        N = len(x)
+
+        alpha, beta, gamma, delta = params
+
+        beta_ab = special.betaln(alpha, beta)
+        beta_gd = special.betaln(gamma, delta)
+
+        indiv_loglike = (special.betaln(alpha + x, beta + T - x) - beta_ab +
+                        special.betaln(gamma, delta + T) - beta_gd)
+
+        recency_T = T - tx - 1
+
+        def _sum(x, tx, recency_T):
+
+            j = np.arange(recency_T+1)
+            return log(
+                np.sum(exp(special.betaln(alpha + x, beta + tx - x + j) - beta_ab +
+                              special.betaln(gamma + 1, delta + tx + j) - beta_gd)))
+
+        for i in np.arange(N):
+            indiv_loglike[i] = logaddexp(indiv_loglike[i],_sum(x[i], tx[i], recency_T[i]))
+
```

Comment:
instead of the for loop, can we just do?
```
indiv_loglike =  logaddexp(indiv_loglike,_sum(x, tx, recency_T))
```
maybe not because of the `_sum` function